packages/datadog-instrumentations/src/body-parser.js

```diff
@@ -1,13 +1,19 @@
 'use strict'
 
+const { AbortController } = require('node-abort-controller') // AbortController is not available in node <15
+const shimmer = require('../../datadog-shimmer')
 const { channel, addHook, AsyncResource } = require('./helpers/instrument')
-
 const bodyParserReadCh = channel('datadog:body-parser:read:finish')
 
-function publishRequestBodyAndNext (request, next) {
+function publishRequestBodyAndNext (req, res, next) {
   return function () {
-    if (bodyParserReadCh.hasSubscribers && request) {
-      bodyParserReadCh.publish({ request })
+    if (bodyParserReadCh.hasSubscribers && req) {
+      const abortController = new AbortController()
+      bodyParserReadCh.publish({ req, res, abortController })
+      if (abortController.signal.aborted) {
+        res.end()
```

Comment:
Is this `res.end()` necessary? if request has been blocked `res.end()` has already been called in `blocking.js`
if it is not, why create an `AbortController` each time?